manifests/configuration.pp

```diff
@@ -0,0 +1,110 @@
+# Manage sysctl variable values.
+#
+# Parameters:
+#  $value:
+#    The value for the sysctl parameter. Mandatory, unless $ensure is 'absent'.
+#  $prefix:
+#    Optional prefix for the sysctl.d file to be created. Default: none.
+#  $ensure:
+#    Whether the variable's value should be 'present' or 'absent'.
+#    Defaults to 'present'.
+#
+# Sample Usage :
+#  sysctl::configuration { 'net.ipv6.bindv6only': value => '1' }
+define sysctl::configuration (
+  String $variable               = $title,
+  String $ensure                 = 'present',
+  Variant[String,Undef] $value   = undef,
+  Variant[String,Undef] $prefix  = undef,
+  String $suffix                 = '.conf',
+  Variant[String,Undef] $comment = undef,
+  Variant[String,Undef] $content = undef,
+  Variant[String,Undef] $source  = undef,
+  Boolean $enforce               = true,
+  String $sysctl_binary          = '/sbin/sysctl',
+  String $sysctl_dir_path        = '/etc/sysctl.d',
+) {
+
+  # If we have a prefix, then add the dash to it
+  if $prefix {
+    $_sysctl_d_file = "${prefix}-${variable}${suffix}"
+  } else {
+    $_sysctl_d_file = "${variable}${suffix}"
```

Comment:
Ah. It's creating a file with the correct name, but the wrong contents. fe15776 on my fork should fix it.

And your behavior is wrong "test-net.ipv6.conf.default.autconf.conf" should _never_ be created because it's not a kernel variable that exists. My fork prevents is intended to prevent creation of files with bad variables and that's exactly why it was failing.